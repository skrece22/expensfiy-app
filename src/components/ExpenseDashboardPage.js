import React from 'react';
import ExpenseList from './ExpenseList';
import ExpeneListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = ()=>(
    <div>
        <ExpeneListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;