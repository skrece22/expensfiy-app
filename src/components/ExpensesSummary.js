import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary= ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedTotal}</h1>
        </div>
    );
}

const mapStateToProps = (state) => {
    const visibileExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expenseCount: visibileExpenses.length,
        expensesTotal: selectExpensesTotal(visibileExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);