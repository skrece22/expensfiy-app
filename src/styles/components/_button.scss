.button{
    background: $blue;
    border: 0;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link{
    background: none;
    border: none;
    color: white;
}

.button--secondary{
    background: #888888;
}