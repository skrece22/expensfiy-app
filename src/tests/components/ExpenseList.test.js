import React from 'React';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render expenses list with correct data',() =>{
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render expenses list with empty expenses',() =>{
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});
