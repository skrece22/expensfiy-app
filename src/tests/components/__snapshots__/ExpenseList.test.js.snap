// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses list with correct data 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render expenses list with empty expenses 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
